libp2p/io/trio.py

```diff
@@ -0,0 +1,33 @@
+import logging
+
+import trio
+from trio import SocketStream
+
+from libp2p.io.abc import ReadWriteCloser
+from libp2p.io.exceptions import IOException
+
+logger = logging.getLogger("libp2p.io.trio")
+
+
+class TrioReadWriteCloser(ReadWriteCloser):
```

Comment:
It *might* be better to do the work to just use the `trio.abc.Stream` API as it seems to encapsulate what you are trying to do here.

https://trio.readthedocs.io/en/latest/reference-io.html#abstract-base-classes